### ISSUE TITLE
Tag formatter documentation

### ISSUE THREAD

**Issue by tcezard** [zestsoftware/zest.releaser#212]
Not sure if this is common knowledge or not but apparently config parser uses interpolation ([issue28935](https://bugs.python.org/issue28935) and [issue20754](https://bugs.python.org/issue20754)) in python3. 
That means the `%` signs are causing the loading to crash.
It can be fixed by using `%%` instead but it did not occur to me when writing the documentation.
Do you thing this should be highlighted ?

**Comment by reinout** [comment]
Bah. Does that mean that, if we use `%%` in the setup.cfg, that zest.releaser should un-escape it?

**Comment by tcezard** [comment]
Nop. It works as expected when you use `%%` which mean ConfigParser already removes it. I just though the documentation might want to explain it.

**Comment by reinout** [comment]
Yes please :-)

**Comment by mauritsvanrees** [comment]
From https://bugs.python.org/issue20754 it looks like it may help if we initialize `ConfigParser` with parameter `interpolation=None`.  Maybe only on Python 3. Can you test if that works?

**Comment by pior** [comment]
This seems to work correctly with Python 3, but fails with for Python 2.7.13

https://travis-ci.org/ludia/pyramid_metrics/jobs/256667732

```shell
$ python setup.py --version
Traceback (most recent call last):
  File "setup.py", line 43, in <module>
    'de = devup.app:cli',
  File "/Users/piorbastida/.pyenv/versions/2.7.13/lib/python2.7/distutils/core.py", line 124, in setup
    dist.parse_config_files()
  File "/Users/piorbastida/.pyenv/versions/2.7.13/lib/python2.7/distutils/dist.py", line 397, in parse_config_files
    val = parser.get(section,opt)
  File "/Users/piorbastida/.pyenv/versions/2.7.13/lib/python2.7/ConfigParser.py", line 623, in get
    return self._interpolate(section, option, value, d)
  File "/Users/piorbastida/.pyenv/versions/2.7.13/lib/python2.7/ConfigParser.py", line 669, in _interpolate
    option, section, rawval, e.args[0])
ConfigParser.InterpolationMissingOptionError: Bad value substitution:
	section: [zest.releaser]
	option : tag-format
	key    : version
	rawval : v%%(version)s
```

**Comment by pior** [comment]
Fix in setuptools: https://github.com/pypa/setuptools/issues/889 

In Py3, setuptools read the config itself, while this doesn't seem to be the case in Py2...

**Comment by Niedes** [comment]
Same here, no matter if I use one or two % signs.
Updating setuptools is not an option, as setuptools greater than 33.1.1 breaks
the buildout of our plone packages. 

`$ python setup.py --version
Traceback (most recent call last):
  File "setup.py", line 61, in <module>
    """,
  File "d:\Python27\lib\distutils\core.py", line 124, in setup
    dist.parse_config_files()
  File "d:\Python27\lib\site-packages\setuptools\dist.py", line 352, in parse_config_files
    _Distribution.parse_config_files(self, filenames=filenames)
  File "d:\Python27\lib\distutils\dist.py", line 397, in parse_config_files
    val = parser.get(section,opt)
  File "d:\Python27\lib\ConfigParser.py", line 623, in get
    return self._interpolate(section, option, value, d)
  File "d:\Python27\lib\ConfigParser.py", line 669, in _interpolate
    option, section, rawval, e.args[0])
ConfigParser.InterpolationMissingOptionError: Bad value substitution:
        section: [zest.releaser]
        option : tag-format
        key    : version
        rawval : %%(version)s`

`$ python setup.py --version
Traceback (most recent call last):
  File "setup.py", line 61, in <module>
    """,
  File "d:\Python27\lib\distutils\core.py", line 124, in setup
    dist.parse_config_files()
  File "d:\Python27\lib\site-packages\setuptools\dist.py", line 352, in parse_config_files
    _Distribution.parse_config_files(self, filenames=filenames)
  File "d:\Python27\lib\distutils\dist.py", line 397, in parse_config_files
    val = parser.get(section,opt)
  File "d:\Python27\lib\ConfigParser.py", line 623, in get
    return self._interpolate(section, option, value, d)
  File "d:\Python27\lib\ConfigParser.py", line 669, in _interpolate
    option, section, rawval, e.args[0])
ConfigParser.InterpolationMissingOptionError: Bad value substitution:
        section: [zest.releaser]
        option : tag-format
        key    : version
        rawval : %(version)s`

**Comment by leorochael** [comment]
I just bumped into this trying to add `tag-format` to `setup.cfg` in `zc.recipe.egg` which shares a repo with `zc.buildout` and needs different tag names.

It doesn't matter if `zest.releaser` can read `tag-format` from `setup.cfg` by tweaking the configparser settings, distutils is going to break (until upstream decides to fix it) since it's using the full recursive interpolation. e.g:

```
661             while depth:  #  Loop through this until it's done
662                 depth -= 1
663                 if value and "%(" in value:
664                     <try to interpolate again>
```

An idea to avoid the issue entirely is to switch the interpolation of the tag-format itself on the `zest.releaser` side to [`.format()`](https://docs.python.org/2/library/stdtypes.html#str.format) based. E.g.:

```
[zest.releaser]
tag-format = v{version}
```

Instead of:

```
[zest.releaser]
tag-format = v%(version)s
```

Of course, this would be a backward incompatible change, so we should think about using a different setting name than `tag-format`.

Unless `zest.releaser` accepts both formats at the same time. Come to think of it, checking whether either one of `%(version)s` or `{version}` is in the format string shouldn't be difficult.

**Comment by reinout** [comment]
Fixed in 9.12.3 by @leorochael (see #232). Thanks!
